### ISSUE TITLE
Version 1.3 on nuget contains DLL wih 1.2.1 version

### ISSUE THREAD

**Issue by bradphelan** [dotnet/corefx#14640]
The issue is detailed on a stackoverflow post but the key issue is that if you download 1.3.0 from the nuget gallery and then check the dll you see that it is really 1.2.1

http://stackoverflow.com/questions/41244162/nuget-seems-to-install-the-wrong-version-of-system-collections-immutable

https://www.nuget.org/packages/System.Collections.Immutable/1.3.0

![image](https://cloud.githubusercontent.com/assets/17650/21355786/a1618b3e-c6cf-11e6-8cba-b566078d8b36.png)

**Comment by bradphelan** [comment]
This means that any packages relying on immutable collection 1.2.* can't use 1.3.0 even though I suspect they are API compatible.

**Comment by ianhays** [comment]
The nuget package version and the assembly version aren't closely tied together. There are plenty of other instances where the package version differs from the assembly version e.g. the System.IO.Compression OOB package. In fact most packages have versions that differ from the version of the assembly that they contain.

**Comment by bradphelan** [comment]
Ok. I'll agree that you don't have to but when you are just shipping a single dll with a nuget package what's the pain in making the version numbers in sync? Normally a CI system just grabs a tag from git, rewrites the assembly info and updates the nuget version to match. It's just one less issue for the consumer of the package to get confused about. Every version of immutable collections so far (that I have tested) has shipped with version numbers in sync apart from 1.3.

But as you say there is no fast rule that they should match so I'll just be aware to check next time. Thanks for the information.

**Comment by ianhays** [comment]
One issue off the top of my head is that when the assembly versions and package versions are the same, getting all of the latest package versions starts to be a hassle. Right now in .net core we've synced all nuget packages to the same version, so if you want the latest of say System.Console and System.Collections, the nuget version will be the same. If we synced it to the assembly version (which will differ), there would be inconsistencies and more difficulty in getting the right packages.

Nuget packages also differ in version based on the release cycle wherein a new release may not also mean a change in the assembly version of a dll.
